File: 1018 - Binary Prefix Divisible By 5.cpp
class Solution {
public:
    vector<bool> prefixesDivBy5(vector<int>& nums) {
        int dec = 0, i = 0;
        vector<bool> res(nums.size());

        for (int n : nums) {
            dec = (dec * 2 + n) % 5;
            res[i++] = (dec == 0);
        }
        return res;
    }
};

File: 1018 - Binary Prefix Divisible By 5.js
/**
 * @param {number[]} nums
 * @return {boolean[]}
 */
var prefixesDivBy5 = function (nums) {
    let res = [], dec = 0;
    for (let n of nums) {
        dec = (dec * 2 + n) % 5;
        res.push(dec == 0);
    }
    return res;
};